### DISCUSSION TITLE
Withdraw unstaked amount bug

### DISCUSSION THREAD

**Discussion by bobyD** [subquery/subql#1048] [Frontier TestNet - Bug Reporting]
After a successful undelegation transaction, this request is displayed on `My profile -> Locked` page. Despite the fact that the undelegation process takes one era in testnet, these tokens have `unlock` status by default. The `Withdraw` button is also active and allows to send an unlimited number of transactions to withdraw tokens from the smart contract. These transactions are successfully performed, but the SQT balance stays the same. But in fact, the withdrawal function starts working only in the next era.

In the attached video, you can see the state of explorer and account balance after multiple Withdrawal transactions in the era where the undelegation request was sent.

https://user-images.githubusercontent.com/69699206/169505679-8e9cd83f-ad47-4724-8ce0-7fc016ea86b5.mp4

`Suggestions`
- To change unlocked assets status from «unlock» to «pending» until the next era comes and to fix the countdown time (now it displays wrong info).
- The «Withdraw» button should become inactive until withdrawal conditions are met. Also, to add the rejection for this type of invalid transactions in the smart contract.

_QuantNode team_

**Comment by Staking7pc** [comment]
We have encountered the same issue and disabling the withdraw button till the actual unlock is possible on the front end seems to be best way to go so the users don't waste transactions on this .

Brightlystake

**Comment by mzxyz** [comment]
@lilian79511 @CaiYiLiang
